src/Draco.Compiler/Api/Compilation.cs

```diff
@@ -201,6 +201,37 @@ public Compilation UpdateSyntaxTree(SyntaxTree? oldTree, SyntaxTree newTree)
             binderCache: null);
     }
 
+    /// <summary>
+    /// Removes the given <paramref name="oldTree"/>.
+    /// </summary>
+    /// <param name="oldTree">The old <see cref="SyntaxTree"/> to delete.
+    /// <returns>A <see cref="Compilation"/> reflecting the change.</returns>
+    public Compilation DeleteSyntaxTree(SyntaxTree oldTree)
```

Comment:
Could we reuse `UpdateSyntaxTree` and make newTree nullable to signal deletion? It already covers creation and update, it would make sense to include that too, since it's 99% ready for that.